changelog.d/14954.misc

```diff
@@ -0,0 +1 @@
+Faster room joins: Refactor internal handling of servers in room to never store an empty list.
```

Comment:
Is this fixing a bug? A potential bug? Or just clean-up?

Reply:
It's somewhere in between fixing a potential bug and a clean up. Probably closer to a clean up.